src/related/related.jsx

```diff
@@ -1,24 +1,52 @@
-import React, { useState, useEffect } from 'react';
+/* eslint-disable camelcase */
+import React, { useEffect, useState } from 'react';
+import axios from 'axios';
+import ProductCard from './productCard.jsx';
 
-function Related(props) {
-  const [count, setCount] = useState(0);
+function Related({ product_id }) {
+  const [products, setProducts] = useState([]);
   useEffect(() => {
-    document.title = `You clicked related ${count} times`;
-  });
-  return (
-    <div>
-      <p>
-        You clicked RELATED
-        {' '}
-        {count}
-        {' '}
-        times
-      </p>
-      <button type="button" onClick={() => setCount(count + 1)}>
-        Click me
-      </button>
-    </div>
-  );
+    axios.get(`/products/${product_id}/related`)
+      .then((res) => res.data)
+      .then((relatedIds) => {
+        Promise.allSettled(relatedIds.map((id) => axios.get(`/products/${id}`)))
+          .then((promisesArr) => promisesArr.map((res) => (res.status === 'fulfilled' ? res.value.data : {})))
+          .then((productsArr) => {
+            Promise.allSettled(relatedIds.map((id) => axios.get(`/products/${id}/styles`)))
+              .then((promisesArr) => promisesArr.map((res) => (res.status === 'fulfilled' ? res.value.data : {})))
+              .then((data) => productsArr.map((product, i) => Object.assign(data[i], product)))
+              .then(finalProductArr => setProducts(finalProductArr));
+          });
+      })
+      .catch((err) => console.log('FAILURE', err));
+  }, []);
```

Comment:
If you change this so that the 2nd argument is `[product_id]`, the cards will updated based on what product is selected.

Reply:
New updates will be on a new merge request